HabitTracker/Program.cs

```diff
@@ -0,0 +1,292 @@
+    var input = Console.ReadLine();
+    int numberInput;
+
+    if (input == "0") GetMenuOption();
+
+    while (!int.TryParse(input, out numberInput))
+    {
+        Console.WriteLine(
+            $"Invalid input format.\n{message}");
+        input = Console.ReadLine();
+    }
+
+    return numberInput;
+}
+
+void DeleteRecord()
+{
+    ViewAllRecords();
+
+    var recordId =
+        GetNumberInput("Please enter the record id number you wish to delete or type 0 to return to the main menu: ");
+
+    using var connection = new SqliteConnection(connectionString);
+    connection.Open();
+    var tableCmd = connection.CreateCommand();
+
+    tableCmd.CommandText =
+        $"DELETE FROM pet_the_dog WHERE Id = '{recordId}'";
+
+    var rowCount = tableCmd.ExecuteNonQuery();
+
+    Console.WriteLine(rowCount == 0
+        ? $"Record {recordId} does not exist in the database. Please try again."
+        : $"\nRecord ID {recordId} has been successfully deleted");
+
+    connection.Close();
+}
+
+void UpdateRecord()
+{
+    ViewAllRecords();
+    var recordId =
+        GetNumberInput("Please enter the record id number you wish to update or type 0 to return to the main menu: ");
+
+    using var connection = new SqliteConnection(connectionString);
+
+    connection.Open();
+
+    var checkCmd = connection.CreateCommand();
+    checkCmd.CommandText = $"SELECT EXISTS(SELECT 1 FROM pet_the_dog WHERE Id = {recordId})";
+    int checkQuery = Convert.ToInt32(checkCmd.ExecuteScalar());
+
+    if (checkQuery == 0)
+    {
+        Console.WriteLine($"\nNo record with Id {recordId} exists.");
+        connection.Close();
+        UpdateRecord();
+    }
+
+
+    var tableCmd = connection.CreateCommand();
+
+    var date = GetDateInput(
+        $"\nPlease enter a date to update record {recordId} (format: yy-mm-dd). Type 0 to return to the main menu: ");
+    var quantity =
+        GetNumberInput(
+            $"Please enter the number of times you pet the dog to update record {recordId} or type 0 to return to the menu: ");
+
+    tableCmd.CommandText =
+        $"""
+                     UPDATE pet_the_dog
+                     SET Quantity = {quantity},
+                         Date = '{date}'
+                     WHERE Id = '{recordId}'
+         """;
+
+    tableCmd.ExecuteNonQuery();
+
+    Console.WriteLine($"Record Id {recordId} has been successfully updated.");
+
+    connection.Close();
+}
+
+void SeedDatabase()
+{
+    Random quantity = new Random();
+
+    for (int i = 0; i < 50; i++)
+    {
+        using var connection = new SqliteConnection(connectionString);
+        connection.Open();
+        var tableCmd = connection.CreateCommand();
+
+        tableCmd.CommandText =
+            $"INSERT INTO pet_the_dog (Date, Quantity) VALUES ('{GenerateSeedDate()}', {quantity.Next(0, 1000)})";
+
+        tableCmd.ExecuteNonQuery();
+        connection.Close();
+    }
+}
+
+string GenerateSeedDate()
+{
+    Random random = new Random();
```

Comment:
Try to only create [one instance](https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-random#avoid-multiple-instantiations) of the `Random` class to make the output more random.

Reply:
Hey @antonderegt, I think I've achieved this by removing the randoms from the seed methods and just initialising the Random class in the Program class. Hope that is what you meant.

Reply:
Yes, that sounds like a good fix!